### DISCUSSION TITLE
Is there a way to reset without triggering watch?

### DISCUSSION THREAD

**Discussion by dninomiya** [react-hook-form/react-hook-form#8852] [Q&A] [answered]
We implement forms that are edited by multiple people in real time. We would like to do the following

1. reflect changes in remote data on local forms in real time
2. reflect local form changes to remote data in real time

And we have tried the following implementation.

```tsx
  useEffect(() => {
    reset(remoteData); // 1. reflect changes in remote data on local forms in real time
  }, [reset, remoteData])

  useEffect(() => {
    return watch(value => {
      postData(value); // 2. reflect local form changes to remote data in real time
    }).unsubscribe
  }, [watch, postData])
```

In this case, `reset` triggers `watch`, so the update runs again and an infinite loop occurs.

Now we prevent infinite loops by checking with deepEqual before updating remote data.

But if you have a smarter approach to the React Hook Form, please let me know.

[Angular forms have an option to not trigger a change event](https://angular.jp/api/forms/FormControl#reset) when the program sets the form values.

A similar option in the React Hook Form reset would be very helpful.

```ts
reset(value, {
  emitEvent: false // Do not trigger watch
})
```

**Comment by bluebill1049** [comment]
you can use `useWatch` and disable the subscription.

**Comment by dninomiya** [comment]
@bluebill1049 
Thank you for your continued support!

I've tried temporarily toggling disabled in useWatch but that didn't work.
https://codesandbox.io/s/react-hook-form-v7-usewatch-forked-p2ygd1?file=/src/index.js

It would be helpful if you could provide a more specific example of use if possible.

Or, do you know example code for a form that allows multiple people to collaborate in real time?

**Reply by Moshyfawn** [reply]
Take a look at the `keepDefaultValues` `reset`  method prop #8233

**Reply by dninomiya** [reply]
This is exactly the thread I was looking for. thank you!

**Reply by Moshyfawn** [reply]
Np 👌

**Comment by dninomiya** [comment, marked as answer]
I finally achieved my goal with the below approach. Thank you for your support! @bluebill1049  @Moshyfawn 

```jsx
function App() {
  const {
    register,
    watch,
    reset,
    formState: { isDirty }
  } = useForm();

  useEffect(() => {
    const subscription = watch((value) => {
      if (isDirty) {
        // #1
        console.log("submitData", value);
      }
    });

    return () => {
      subscription.unsubscribe();
    };
  }, [watch, isDirty]);

  const fetchRemoteData = () => {
    console.log("fetch remoteData");
    const mockRemoteData = {
      firstName: "foo",
      lastName: "baa"
    };

    // Hoping to skip #1(submitData)
    reset(mockRemoteData, {
      isDirty: false // this was the point
    });
  };

  return (
    <form>
      <input {...register("firstName")} placeholder="firstName" />
      <input {...register("lastName")} placeholder="lastName" />
      <button type="button" onClick={fetchRemoteData}>
        fetch remote data
      </button>
    </form>
  );
}
```

https://codesandbox.io/s/react-hook-form-v7-usewatch-forked-f9ethv?file=/src/index.js:144-1040

I will refer to [this comment ](https://github.com/react-hook-form/react-hook-form/discussions/8233#discussioncomment-2621207)if I need more precise synchronization with remote data.

**Comment by khanakia** [comment]
How to achieve the same with newer versions as there is no `subscription.unsubscribe();` function does exists
